src/Shared/FileUtilities.cs

```diff
@@ -223,6 +223,20 @@ internal static string EnsureNoTrailingSlash(string path)
             return path;
         }
 
+        /// <summary>
+        /// Ensures the path does not have a leading or trailing slash.
+        /// </summary>
+        internal static string EnsureNoLeadingOrTrailingSlash(string path)
+        {
+            if (String.IsNullOrEmpty(path))
+            {
+                return path;
+            }
+            int start = IsSlash(path[0]) ? 1 : 0;
+            int end = IsSlash(path[path.Length - 1]) ? 1 : 0;
+            return path.Length - start - end > 0 ? path.Substring(start, path.Length - start - end) : String.Empty;
```

Comment:
Why not reuse [String.Trim(char[])]([https://docs.microsoft.com/en-us/dotnet/api/system.string.trim?view=netcore-3.1#System_String_Trim_System_Char](https://docs.microsoft.com/en-us/dotnet/api/system.string.trim?view=netcore-3.1#System_String_Trim_System_Char)___) to implement this?

Reply:
That's actually really nice. Thanks for the suggestion!

Reply:
`String.Trim()` removes all leading/trailing characters from the set, though. This behavior matches the name of the method but it's not what the method was doing. Is it guaranteed to not break anything?

Example: "\\hello"
Previously it would become "\hello", now after calling `String.Trim()` it is "hello".

Reply:
I'm reasonably convinced that if there were multiple slashes on either side, that was a mistake because this is used for a path; there's no reason (unless you're explicitly trying to hack around a flawed implementation of EnsureNoLeadingSlash) to have multiple slashes in a row.

Additionally, to comment on the fact that I no longer escaped projectDirectoryNoRoot: it was already escaped just above there, so I believe that was extra work to no benefit. If there is a reason that double-escaping something (and presumably double-unescaping something) is preferred, I can revert that part.

Reply:
Got it, thank you. How about the slash normalization done in `FixFilePath()` and called from `EnsureNoLeadingSlash()` and `EnsureNoTrailingSlash()`? Is it also something we know for sure wasn't needed?

Reply:
I am not confident about that and would actually guess that might be problematic, although the fact that it passed tests suggests it might be ok. To be safe, I'll add it back. Good catch!

Reply:
And while I'm at it, I'm not sure why I should have to substring, then trim. Why not both at once?

Reply:
Actually, never mind. I'd looked at this and forgotten: Path.GetPathRoot does slash normalization, so I don't need to. I'll still put EnsureNoLeadingOrTrailingSlash back so I can do substring at the same time.

Reply:
Another question: why use the root path to calculate the length of the root path, then use that to substring something else instead of just using the root path in the first place? Very confusing.